virtcontainers/container_test.go

```diff
@@ -255,10 +261,9 @@ func TestContainerAddDriveDir(t *testing.T) {
 		t.Fatalf("Error with hotplugDrive :%v", err)
 	}
 
-	if container.state.Fstype == "" || !container.state.HotpluggedDrive {
+	if container.state.Fstype != "" && container.state.HotpluggedDrive {
```

Comment:
So you are actually checking here that the drive is not hotplugged, which changes what the test is supposed to verify.
I think you need to add a noopAgent for the pod and have the noopAgent advertise the block support for the test.